doc/codingGuidelines.md

```diff
@@ -0,0 +1,229 @@
+#define PI_ROUNDED 3.0
+```
+
+**Schema Names**
+<Not yet defined, proposals are welcome>
+
+### Documentation (class, method, variable, comments)
+* [Doxygen ](http://www.doxygen.nl/index.html) will be used to generate documentation from mayaUsd C++ sources.
+* Doxygen tags must be used to document classes, function parameters, function return values, and thrown exceptions.
+* The mayaUsd project does require the use of any Doxygen formatting style ( [Doxygen built-in formatting](http://www.doxygen.nl/manual/commands.html) )
+* Comments for users of classes and functions must be written in headers files. Comments in definition files are meant for contributors and maintainers.
+
+### Include directive
+All included public header files from outside and inside the project should be `#include`’d using angle brackets. For example:
+```cpp
+#include <pxr/base/tf/stringUtils.h>
+#include <mayaUsd/nodes/stageData.h>
+```
+
+Private project’s header files should be `#include`'d using the file name when in the same folder. Private headers may live in sub-directories, but they should never included using "._" or ".._" as part of a relative paths. For example:
+```cpp
+#include “privateUtils.h"
+#include “pvt/helperFunctions.h"
+```
+
+### Include order
+Headers should be included in the following order, with each section separated by a blank line and files sorted alphabetically:
+
+1. Related header
+2. C system headers
+3. C++ standard library headers
+4. Other libraries’ headers
+5. Autodesk + Maya headers
+6. Pixar + USD headers
+7. Your project’s headers
+8. Conditional includes
+
+```cpp
+#include “exportTranslator.h"
+ 
+#include <string>
+ 
+#include <maya/MFileObject.h>
+#include <maya/MGlobal.h>
+#include <maya/MSelectionList.h>
+#include <maya/MString.h>
+
+#include <mayaUsd/fileio/jobs/jobArgs.h>
+#include <mayaUsd/fileio/jobs/writeJob.h>
+#include <mayaUsd/fileio/shading/shadingModeRegistry.h>
+#include <mayaUsd/fileio/utils/writeUtil.h>
+
+#if defined(WANT_UFE_BUILD)
+  #include <ufe/ufe.h>
+#endif
+```
+
+### Cross-platform development
+* Consult the [build.md](https://github.com/Autodesk/maya-usd/blob/dev/doc/build.md) compatibility table to ensure you use the recommended tool (i.e., dcompiler, operating system, CMake, etc.) versions.
+* Before pull requests (PRs) are considered for integration, they must compile and all tests should pass on all suppported platforms.
+
+### Conditional compilation (Maya, USD, UFE version)
+**Maya**
+	* MAYA_API_VERSION is the consistent macro to test Maya version (MAYA_APP_VERSION * 10000 + MAJOR_VERSION * 100 + MINOR_VERSION)
+	* MAYA_APP_VERSION is available only since Maya 2019 and is a simple year number, so it is not allowed.
+**UFE**
+	* WANT_UFE_BUILD equals 1 if UFE is found and it should be used for conditional compilation on codes depending on UFE.
+
+**USD**
+	* USD_VERSION_NUM is the macro to test USD version (USD_MAJOR_VERSION * 10000 + USD_MINOR_VERSION * 100 + USD_PATCH_VERSION)
```

Comment:
It might be good to wrap all these version variable names in backticks to make them stand out a little better in the rendered text.